base/system/sys_info_starboard.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/system/sys_info_starboard.h"
+
+#include "build/build_config.h"
+
+#if BUILDFLAG(IS_ANDROID)
+#include <sys/system_properties.h>
+#endif  // BUILDFLAG(IS_ANDROID)
+
+namespace base {
+namespace starboard {
+
+namespace {
+
+const char kUnknownValue[] = "unknown";
+
+}  // namespace
+
+#if BUILDFLAG(IS_ANDROID)
+std::string SbSysInfo::OriginalDesignManufacturer() {
+  char original_design_manufacturer_str[PROP_VALUE_MAX];
+  __system_property_get("ro.product.manufacturer", original_design_manufacturer_str);
+  return std::string(original_design_manufacturer_str);
+}
+
+std::string SbSysInfo::ChipsetModelNumber() {
+  char chipset_model_number_str[PROP_VALUE_MAX];
+  __system_property_get("ro.board.platform", chipset_model_number_str);
+  return std::string(chipset_model_number_str);
+}
+
+std::string SbSysInfo::ModelYear() {
+  char model_year_str[PROP_VALUE_MAX];
+  __system_property_get("ro.oem.key1", model_year_str);
+  if (strcmp(model_year_str, kUnknownValue) == 0 || strlen(model_year_str) < 10) {
```

Comment:
Move `kUnknownValue` definition to here, and make it a constexpr std::string, then use the normal std::string comparisons plz.

Reply:
std::string is non-trivial and cannot be made constexpr, update it to const instead.